object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -121,29 +123,52 @@ void checkMustCallInvoked(ControlFlowGraph cfg) {
           handleReturn((ReturnNode) node, cfg, newDefs);
         } else if (node instanceof MethodInvocationNode || node instanceof ObjectCreationNode) {
           handleInvocation(newDefs, node);
-        } else if (node instanceof TypeCastNode) {
-          handleTypeCast((TypeCastNode) node, newDefs);
+        } else if (node instanceof TypeCastNode || node instanceof TernaryExpressionNode) {
+          handleTypeCastOrTernary(node, newDefs);
         }
       }
 
       handleSuccessorBlocks(visited, worklist, newDefs, curBlockLocals.block);
     }
   }
 
-  private void handleTypeCast(TypeCastNode node, Set<ImmutableSet<LocalVarWithTree>> defs) {
-    Node operand = node.getOperand();
-    if (operand instanceof MethodInvocationNode || operand instanceof ObjectCreationNode) {
-      if (!shouldSkipInvokePseudoAssignCheck(operand, defs)) {
-        checkPseudoAssignToOwning(node);
-      }
+  private void handleTypeCastOrTernary(Node node, Set<ImmutableSet<LocalVarWithTree>> defs) {
+    if (!(node instanceof TypeCastNode || node instanceof TernaryExpressionNode)) {
+      throw new BugInCF("unexpected node type " + node.getClass());
+    }
+    if (!nestedInCastOrTernary(node) && hasNestedPseudoAssignInvoke(node, defs)) {
+      checkPseudoAssignToOwning(node);
```

Comment:
Can we replace hasNestedPseudoAssignInvoke() with !shouldSkipInvokePseudoAssignCheck()? I can't understand why we need to check TypeCastNode and TernaryExpressionNode in hasNestedPseudoAssignInvoke() again.

Reply:
@Nargeshdb my understanding was that `shouldSkipInvokePseudoAssignCheck()` is intended to work for nodes that are invocations.  Here, `node` is either a `TypeCastNode` or a `TernaryExpressionNode`.  So, `hasNestedPseudoAssignInvoke()` does the work of recursing through the operands of `node` looking for invocation nodes, and then passing those nodes to `shouldSkipInvokePseudoAssignCheck()`.  If _any_ of the nested invokes requires a pseudo-assign check, `hasNestedPseudoAssignInvoke()` will return true.  Does that sound right?  We do not want to do a pseudo-assign check if there are no nested invokes.  E.g., we don't want to do the check for `b ? 3 : -3`.

Reply:
In 30b5eae I added a test that shows the need for `hasNestedPseudoAssignInvoke()`